### ISSUE TITLE
AugmentedManifest File not Working for Image Classification Built-In

### ISSUE THREAD

**Issue by canada4663** [awslabs/amazon-sagemaker-examples#530]
I followed the example below for setting up an AugmentedManifest file training for the built-in image classification library:

https://github.com/awslabs/amazon-sagemaker-examples/blob/1e4a0d21ebf5731aae900aecdeff61dfa8c65a0c/ground_truth_labeling_jobs/object_detection_augmented_manifest_training/object_detection_augmented_manifest_training.ipynb

I repeatedly get the error shown below:

```
2018-12-16 16:39:16 Algorithm Error: pos_cls_weights (caused by KeyError)
[12/16/2018 16:39:16 ERROR 140380304340800] Algorithm Error: pos_cls_weights (caused by KeyError)
```
 
Here is my training param setup, which with the exception of the AugmentedManifest file InputDataConfig, is the exact same as a working job using the LST file mechanism:

```
training_params = \
{
    # specify the training docker image
    "AlgorithmSpecification": {
        "TrainingImage": training_image,
        "TrainingInputMode": "Pipe"
    },
    "RoleArn": role,
    "OutputDataConfig": {
        "S3OutputPath": 's3://{}/{}/output'.format(bucket, job_name_prefix)
    },
    "ResourceConfig": {
        "InstanceCount": 1,
        "InstanceType": "ml.p2.8xlarge",
        "VolumeSizeInGB": 50
    },
    "TrainingJobName": job_name,
    "HyperParameters": {
        "image_shape": image_shape,
        "num_layers": str(num_layers),
        "num_training_samples": str(num_training_samples),
        "augmentation_type": str(augmentation_type),
        "num_classes": str(num_classes),
        "mini_batch_size": str(mini_batch_size),
        "epochs": str(epochs),
#        "kv_store": str(kv_store),
        "learning_rate": str(learning_rate),
        "lr_scheduler_factor": str(lr_scheduler_factor),
        "lr_scheduler_step": str(lr_scheduler_step),
        "optimizer": str(optimizer),
        "eps": str(eps),
        "beta_1": str(beta_1),
        "beta_2": str(beta_2),
        "top_k": str(top_k),
        "resize": str(resize),
        "checkpoint_frequency": str(checkpoint_frequency),
        "use_pretrained_model": str(use_pretrained_model),
        "use_weighted_loss": str(use_pretrained_model),
        "early_stopping": str(early_stopping),
        "early_stopping_min_epochs": str(early_stopping_min_epochs),
        "early_stopping_patience": str(early_stopping_patience),
        "early_stopping_tolerance": str(early_stopping_tolerance),
    },
    "StoppingCondition": {
        "MaxRuntimeInSeconds": 360000
    },
#Training data should be inside a subdirectory called "train"
#Validation data should be inside a subdirectory called "validation"
#The algorithm currently only supports fullyreplicated model (where data is copied onto each machine)
    "InputDataConfig": [
        {
            "ChannelName": "train",
            "DataSource": {
                "S3DataSource": {
                    "S3DataType": "AugmentedManifestFile",
                    "S3Uri": trainManifest,
                    "S3DataDistributionType": "FullyReplicated",
                    'AttributeNames': ["source-ref","class"]
                }
            },
#            "InputMode": 'Pipe',
            "ContentType": "application/x-recordio",
            "RecordWrapperType": "RecordIO",
            "CompressionType": "None"
        },
        {
            "ChannelName": "validation",
            "DataSource": {
                "S3DataSource": {
                    "S3DataType": "AugmentedManifestFile",
                    "S3Uri": validateManifest,
                    "S3DataDistributionType": "FullyReplicated",
                    'AttributeNames': ["source-ref","class"]
                }
            },
#            "InputMode": 'Pipe',
            "ContentType": "application/x-recordio",
            "RecordWrapperType": "RecordIO",
            "CompressionType": "None"
        }
    ]
}
```
 
I have tried a number of permutations around S3DataDistributionType, RecordWrapperType, InputMode, and AttributeNames.  All result in the same error.  Hoping its something silly I'm missing.

Thanks!

**Comment by gurumurthys** [comment]
You are setting  the use_weighted_loss parameter - "use_weighted_loss": str(use_pretrained_model)". This is used only for the multi-label setup. Can you try without this parameter?

**Comment by canada4663** [comment]
Wow.  This is running, sorry for the miss on my part.  Oddly, this doesn't fail the model unless in AugmentedManifest mode.

Will let the job finish training and then provided all is well, will close the issue (unless you want to do more error checking / warning in the config)
